### DISCUSSION TITLE
[Feature request] Wrapping custom snippet around text

### DISCUSSION THREAD

**Discussion by ukinaa** [gsantner/markor#2059] [Ideas] [answered]
@harshad1 I'd love to have a custom snippet that wraps around selected text, like with built-in ActionButtons (e.g. bold puts ** around a selection)? Currently, this is not the case: if I first select some text and then select my snippet, the text gets replaced by the snippet.

PS. On my Linux desktop I'm using corresponding feature in Sublime Text. Below is an example (that wraps HTML comment  around selected text) from that system, but I've no idea what's happening under the hood.

```
<snippet>
    <content><![CDATA[
<!--${SELECTION/\n/<\/li>\n<li>/g}-->
]]></content>
    <description>Hide</description>
    <scope>text.html</scope>
</snippet>
```

**Comment by harshad1** [comment, marked as answer]
Please  look at + test https://github.com/gsantner/markor/pull/2081 @ukinaa
